### PR TITLE
Provide a way to override condarc search path: `CONDA_NO_CONFIG_SEARCH_PATH`

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Some uses of `conda` require working in absolute isolation from user configurations. One example is conda-standalone as used in `constructor`.

Some issues originating from this problem can be found here: https://github.com/conda/constructor/issues/568. 

The solution is simple: in the presence of a non-empty `NO_CONDARC` environment variable, the default search path is overridden and thus no configuration is loaded from disk, at all. 

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [X] Add / update necessary tests?
- [X] Add / update outdated documentation?

Closes https://github.com/conda/constructor/issues/542

Questions: 

* An alternative solution to this problem would be exposing `CONDA_SEARCH_PATH` as a configurable list. Maybe not linked directly in the context object (bootstrapping issue). Instead of using `NO_CONDARC`, the user could set `CONDA_SEARCH_PATH=""`, achieving the same effect. This would be more flexible but also confusing and unneeded?

### CODE REVIEW

**Comment by chenghlee** [comment]
Would setting this override the system-/install-level configuration files (i.e., those in `$CONDA_ROOT`)? I'm know sysadmins who use those configuration files to, e.g., constrain what channels users can access, and it's probably not a good idea to give users a mechanism to override that.

**Comment by kenodegard** [comment]
How about `CONDA_CONFIG_SEARCH_PATH`? This would perhaps be more clear to those unfamiliar with the history of "rc"?

Something to consider, git offers a [`GIT_CONFIG_NOSYSTEM` boolean environment variable](https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables). Not sure how much value there is in exposing the `SEARCH_PATH` versus a boolean.

File: tests/base/test_context.py [hunk 1]
```diff
@@ -547,3 +547,30 @@ def test_validate_prefix_name(prefix, allow_base, mock_return_values, expected):
...
+                channels: ["uniquely_named_channel"]
+                """
+            )
+        )
+    # $CONDARC allows us to add one more file or directory to the search path
+    with env_var("CONDARC", custom_condarc):
+        reset_context()
+        assert "uniquely_named_channel" in context.channels
+
+        # $CONDA_RC_SEARCH_PATH overrides the default search path, but
```
Comment (pavelzw) [inline tests/base/test_context.py:567]:
```suggestion
        # $CONDA_RC_SEARCH_PATH overrides the default search path,
```

**Comment by jaimergp** [comment]
Hm, I keep thinking about this and overriding the default search path with full flexibility might incur into a support team asking "have you changed the default search path? what is the value?", which is something I don't think it's needed.

So... inspired by your suggestion, @kenodegard, the last approach is this one:
* Users can set `CONDA_NO_CONFIG_SEARCH_PATH` to change the default search path to just `$CONDARC`.
* If they _also_ set `CONDARC` to a file or directory, this will be loaded.
* With no `CONDARC` env var, no file is loaded.

This allows people to override the default condarc, if needed, with a custom one, but also prevents issues coming from misconfiguring `CONDA_SEARCH_PATH`.

Does this look like a compromise?

**Comment by pavelzw** [comment]
Sounds good to me 👍🏻

**Comment by jaimergp** [comment]
Turns out this has been attempted before. See prior discussion: https://github.com/conda/conda/issues/6902.

There were concerns about the "security model" for sysadmins.

This might live as a perma-patch on conda-standalone for constructor purposes.

**Pull request closed without merging**